lib/manageiq/api/client/collection.rb

```diff
@@ -42,6 +48,71 @@ def self.subclass(name)
             const_set(klass_name, klass)
           end
         end
+
+        def search(mode, options)
+          options[:limit] = 1 if mode == :first
+          result = get(parameters_from_query_relation(options))
+          case mode
+          when :first then result.first
+          when :last  then result.last
+          when :all   then result
+          else raise "Invalid mode #{mode} specified for search"
+          end
+        end
+
+        private
+
+        def parameters_from_query_relation(options)
+          api_params = {}
+          [:offset, :limit].each { |opt| api_params[opt] = options[opt] if options[opt] }
+          api_params[:attributes] = options[:select].join(",") if options[:select].present?
+          if options[:where]
+            api_params[:filter] ||= []
+            api_params[:filter] += filters_from_query_relation("=", options[:where])
+          end
+          if options[:not]
+            api_params[:filter] ||= []
+            api_params[:filter] += filters_from_query_relation("!=", options[:not])
+          end
+          if options[:order]
+            order_parameters_from_query_relation(options[:order]).each { |param, value| api_params[param] = value }
+          end
+          api_params
+        end
+
+        def filters_from_query_relation(condition, option)
+          filters = []
+          option.each do |attr, values|
+            Array(values).each do |value|
+              value = "'#{value}'" if value.kind_of?(String) && !value.match(/^(NULL|nil)$/i)
```

Comment:
It's just that the ' ' or " " declaration gives the users (and us) better control on type, i.e. differentiating string vs. let's say numeric (we'd drive arel to do string match or numeric equality),  something != '5' vs something != 5, also on server side we can explicitly check for nil (ssui checks some attributes for nil value for example, so doing something='nil' (a string with value "nil") vs something=nil  which really tells us to check for nil.